referenceframe/frame_json.go

```diff
@@ -87,6 +94,18 @@ func (cfg *LinkConfig) ParseConfig() (*LinkInFrame, error) {
 	return NewLinkInFrame(cfg.Parent, pose, cfg.ID, geom), nil
 }
 
+// Rename makes a deep copy of the config with an ID corresponding to the given name.
+func (cfg *LinkConfig) Rename(name string) *LinkConfig {
```

Comment:
`Rename` is a misleading name here; it makes one think they're mutating the underlying struct.

Would prefer something like `CopyWithNewName`